components/CreateCycle/CreateCycle.jsx

```diff
@@ -45,12 +45,30 @@ export default ({ navigation, route }) => {
   const [workouts, setWorkouts] = React.useState(cycleDetails);
   const dispatch = useDispatch();
 
+  const cycles = useSelector((state) => state.cycles);
   const Stack = createStackNavigator();
 
   function updateOrder(workoutList) {
     setWorkouts(workoutList);
   }
 
+  function removeWorkout(workoutName) {
+    if (workouts.length === 1) {
+      alert('Cannot delete last workout');
+      return;
+    }
+    for (let i = 0; i < workouts.length; i += 1) {
+      if (workouts[i].name === workoutName) {
+        workouts.splice(i, 1);
+      }
+    }
+    if (!isNewCycle) {
+      // Makes sure selectedCycleIndex decreases if number of workouts in cycle decreases
+      dispatch(actions.cycles.decrementSelectedCycleIndex(cycles.selectedCycleIndex, cycleDetails.length));
+    }
+    updateOrder(workouts);
+  }
```

Comment:
Couple comments on this function:
1) I think you could simplify this function a bit if you just took in the index of the workout instead of the workoutName, you wouldn't need to do the for loop in the middle
2) Also, correct me if I'm wrong, but this code would decrement the selected cycle index if the cycle is being edited, not just if it was the selected cycle.